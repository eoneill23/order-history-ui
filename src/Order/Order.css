.order {
  align-items: center;
  background-color: white;
  box-shadow: 5px 10px;
  display: flex;
  justify-content: space-between;
  margin: 20px 0 20px 0;
  width: 90%;
}

.order-img {
  height: 100px;
  width: 100px;
}

.price {
  font-weight: bold;
}