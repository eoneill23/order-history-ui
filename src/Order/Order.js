import React from 'react';
import './Order.css'

const Order = ({ id, img, name, description, price, deleteOrder }) => {

  return (
    <article className='order'>
      <img src={img} alt='Image of a placed order' className='order-img'></img>
      <div className='content-wrapper'>
        <h3>{name}</h3>
        <p>{description}</p>
      </div>
      <div className='button-wrapper'>
        <p className='price'>${price}</p>
        <button className='order-btn' onClick={() => deleteOrder(id)}>Remove From History</button>
      </div>
    </article>
  )
}

export default Order;