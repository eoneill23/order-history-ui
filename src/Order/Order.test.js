import React from 'react';
import { shallow } from 'enzyme';
import Order from './Order.js';

describe('Order', () => {
  let wrapper, deleteOrderMock;

  beforeEach(() => {
    deleteOrderMock = jest.fn();
    wrapper = shallow(<Order id={1} img={'twitter.com'} name={'stereo'} description={'cool stereo'} price={'200'} deleteOrder={deleteOrderMock}/>)
  });

  it('should render without crashing', () => {
    expect(wrapper.length).toEqual(1)
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire the deleteOrder method when the button is clicked', () => {
    wrapper.find('button').simulate('click');

    expect(deleteOrderMock).toHaveBeenCalled();
  });
});