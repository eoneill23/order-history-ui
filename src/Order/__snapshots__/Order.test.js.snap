// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should match the snapshot with all data passed in correctly 1`] = `
<article
  className="order"
>
  <img
    alt="Image of a placed order"
    className="order-img"
    src="twitter.com"
  />
  <div
    className="content-wrapper"
  >
    <h3>
      stereo
    </h3>
    <p>
      cool stereo
    </p>
  </div>
  <div
    className="button-wrapper"
  >
    <p
      className="price"
    >
      $
      200
    </p>
    <button
      className="order-btn"
      onClick={[Function]}
    >
      Remove From History
    </button>
  </div>
</article>
`;
